server/v2/cometbft/types/types.go

```diff
@@ -21,3 +22,15 @@ type ConsensusInfo struct { // TODO: this is a mock, we need a proper proto.Mess
 	proto.Message
 	corecomet.Info
 }
+
+type Store interface {
+	store.Store
+	Query(storeKey string, version uint64, key []byte, prove bool) (QueryResult, error)
+}
+
+type QueryResult interface {
+	Key() []byte
+	Value() []byte
+	Version() uint64
+	Proof() interface{} // CommitmentOp // TODO: use correct type
```

Comment:
ics23 proofOp

Reply:
In store we use:

```
type CommitmentOp struct {
	Type  string
	Key   []byte
	Spec  *ics23.ProofSpec
	Proof *ics23.CommitmentProof
}
```

Where type is always `"ics23:iavl"` and key is the same as the query result key. So I'll add ics23.ProofSpec and ics23.CommitmentProof separately